flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/parse/HiveParserDDLSemanticAnalyzer.java

```diff
@@ -1494,26 +1848,95 @@ private Serializable analyzeAlterTableAddParts(
         }
 
         // add the last one
-        if (currentPart != null) {
-            addPartitionDesc.addPartition(currentPart, currentLocation);
+        if (currentPartSpec != null) {
+            specs.add(new CatalogPartitionSpec(currentPartSpec));
+            Map<String, String> props = new HashMap<>();
+            if (currentLocation != null) {
+                props.put(TABLE_LOCATION_URI, currentLocation);
+            }
+            partitions.add(new CatalogPartitionImpl(props, null));
         }
 
-        return new DDLWork(getInputs(), getOutputs(), addPartitionDesc);
+        ObjectIdentifier tableIdentifier =
+                tab.getDbName() == null
+                        ? parseObjectIdentifier(tab.getTableName())
+                        : catalogManager.qualifyIdentifier(
+                                UnresolvedIdentifier.of(tab.getDbName(), tab.getTableName()));
+        CatalogBaseTable catalogBaseTable = getCatalogBaseTable(tableIdentifier);
+        if (catalogBaseTable instanceof CatalogView) {
+            throw new ValidationException("ADD PARTITION for a view is not supported");
+        }
+        return new AddPartitionsOperation(tableIdentifier, ifNotExists, specs, partitions);
     }
 
-    // Get the partition specs from the tree
-    private List<Map<String, String>> getPartitionSpecs(CommonTree ast) {
-        List<Map<String, String>> partSpecs = new ArrayList<>();
-        // get partition metadata if partition specified
-        for (int childIndex = 0; childIndex < ast.getChildCount(); childIndex++) {
-            HiveParserASTNode partSpecNode = (HiveParserASTNode) ast.getChild(childIndex);
-            // sanity check
-            if (partSpecNode.getType() == HiveASTParser.TOK_PARTSPEC) {
-                Map<String, String> partSpec = getPartSpec(partSpecNode);
-                partSpecs.add(partSpec);
+    private Operation convertAlterViewProps(String tableName, Map<String, String> newProps) {
+        ObjectIdentifier viewIdentifier = parseObjectIdentifier(tableName);
+        CatalogBaseTable baseTable = getCatalogBaseTable(viewIdentifier);
+        CatalogView oldView = (CatalogView) baseTable;
+        Map<String, String> props = new HashMap<>(oldView.getOptions());
+        props.putAll(newProps);
+        CatalogView newView =
+                new CatalogViewImpl(
+                        oldView.getOriginalQuery(),
+                        oldView.getExpandedQuery(),
+                        oldView.getSchema(),
+                        props,
+                        oldView.getComment());
+        return new AlterViewPropertiesOperation(viewIdentifier, newView);
+    }
+
+    private void checkAlterTableLegal(String tableName, boolean expectView) {
```

Comment:
We can let this method return the old table/view and reuse it in subsequent methods. Getting the table involves traffic to HMS so we should restrain the number of such calls.